store/generated_key/memstore/v2/memstore.go

```diff
@@ -0,0 +1,219 @@
+package memstore
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"math/big"
+
+	"github.com/consensys/gnark-crypto/ecc/bn254"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/rlp"
+
+	"github.com/Layr-Labs/eigenda-proxy/common"
+	"github.com/Layr-Labs/eigenda-proxy/store/generated_key/memstore/ephemeral_db"
+	"github.com/Layr-Labs/eigenda-proxy/store/generated_key/memstore/memconfig"
+	"github.com/Layr-Labs/eigenda/api/clients/codecs"
+	"github.com/Layr-Labs/eigenda/api/clients/v2/verification"
+
+	cert_verifier_binding "github.com/Layr-Labs/eigenda/contracts/bindings/EigenDACertVerifier"
+
+	"github.com/Layr-Labs/eigensdk-go/logging"
+)
+
+const (
+	BytesPerFieldElement = 32
+)
+
+// randomBytes ... Generates random byte slice provided
+// size. Errors when generating are ignored since this is only
+// used for constructing dummy certificates when testing insecure integrations.
+// in the worst case it doesn't work and returns empty arrays which would only
+// impact memstore operation in the event that two identical payloads are provided
+// since they'd resolve to the same commitment and blob key. This shouldn't matter
+// given this is typically used for testing standard E2E functionality against a rollup
+// stack which SHOULD never submit an identical batch more than once.
```

Comment:
idk if this is true... if l2 reorg happens same batch will be resubmitted on op stack afair

Reply:
Yes but this an extreme edge case which assumes that random fails for whatever reason